src/core/matter/mod.rs

```diff
@@ -112,15 +108,15 @@ impl Matter {
     }
 
     pub fn code(&self) -> String {
-        self.code.to_string()
+        self.code.clone()
```

Comment:
I feel we made the inverse of this change in a previous PR

Reply:
This is actually already a string, the code. We need to clone it when someone requests it from us, unless we return a ref. If we return a ref we'd need to be very careful in the calling code not to clobber something inside an opaque struct, was my reasoning. The to_string() was serving as a clone(), and I feel this is clearer since the underlying data is a `String`.

Reply:
Thanks for the explanation, makes sense.